### PR TITLE
[Spike] Server side copy of blob - extended evaluation

### PR DESCRIPTION
**Pull request by ouphi**
# Server side copy of blob
- Clarified the best way to copy blobs on server side with an Azure Storage developer.
- Documented the outcome:
    - Explained the most optimal way to copy blobs on server side with java SDK.
    - Added code snippets and tested it.
# ADR
Created simple ADR explaining the approach to copy blobs on server side.

### CODE REVIEW

File: blobspike/ADR.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Server side copy extension
+
+## Copy with Java SDK
+
+The optimal strategy to copy blobs on server side is to copy the blob by blocks in parallel.
+The size of the block is configurable. The parallelization is not handled by the client but should be handled the logic on top of it.
+In order to achieve that, call [stageBlockFromUrl](https://docs.microsoft.com/en-us/java/api/com.azure.storage.blob.specialized.blockblobclient.stageblockfromurl?view=azure-java-stable) for each block to copy.
+Once blocks are successfully written, call [commitBlockList](https://docs.microsoft.com/en-us/java/api/com.azure.storage.blob.specialized.blockblobclient.commitblocklist?view=azure-java-stable).
+This will create the blob composed ot the blocks.
```
Comment (cpeeyush) [inline blobspike/ADR.md:9]:
```suggestion
This will create the blob composed of the blocks.
```

**Review by algattik — APPROVED** [review]
Can we some indication of which block size to use? Is something preset in DML/azcopy?

**Review by Izzzu — APPROVED** [review]

**Comment by Izzzu** [comment]
> Can we some indication of which block size to use? Is something preset in DML/azcopy?

Yes, azcopy uses blocks of ~8 MB, but we didn't find any recommendation in the docs.

**Comment by Izzzu** [comment]
> Can we some indication of which block size to use? Is something preset in DML/azcopy?

@algattik  Yes, azcopy used blocks of ~8 MB, but we didn;t find any recommendation in the documentation.

**Pull request merged**